src/test/java/test/methodselection/MethodSelectionTest.java

```diff
@@ -0,0 +1,65 @@
+package test.methodselection;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.testng.annotations.Test;
+import org.testng.xml.XmlClass;
+import org.testng.xml.XmlSuite;
+import org.testng.xml.XmlTest;
+import test.InvokedMethodNameListener;
+import test.SimpleBaseTest;
+
+public class MethodSelectionTest extends SimpleBaseTest {
```

Comment:
would this have been simpler, if we resorted to using a data provider ? Something like this ?

```java
  @Test(dataProvider = "dp")
  public void testMethod(String methodName, String expected) {
    XmlSuite xmlSuite = createXmlSuite("Dollar method Suite");
    XmlTest xmlTest = createXmlTest(xmlSuite, "Dollar method Test");
    XmlClass xmlClass = createXmlClass(xmlTest, DollarMethodSample.class);
    createXmlInclude(xmlClass, methodName);
    InvokedMethodNameListener listener = run(xmlSuite);
    assertThat(listener.getInvokedMethodNames()).containsExactly(expected);
  }

  @DataProvider(name ="dp")
  public Object[][] getData() {
    return new Object[][] {
            {"$", "$"},
            {"another_test", "another_test"},
            {"another$_test", "another$_test"},
            {"another\\$_test\\$$", "another$_test$"},
    };
  }
```